src/webapp/hooks/useSpecificCall.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+import { CompositionRoot } from "../../CompositionRoot";
+import { CallStatusTypes } from "../../domain/entities/GlassCallStatus";
+import { GlassState } from "./State";
+
+type GlassCallState = GlassState<CallStatusTypes>;
+
+export function useSpecificCall(compositionRoot: CompositionRoot, moduleId: string, orgUnit: string, period: number) {
+    const [call, setCall] = useState<GlassCallState>({
+        kind: "loading",
+    });
+
+    useEffect(() => {
+        compositionRoot.glassCall.getSpecificCall(moduleId, orgUnit, period).run(
+            currentCall => {
+                //If calls are filtered on module && orgunit && period, then only one call should be returned,
```

Comment:
This logic seems an application domain logic. It's an automatization logic.

The application domain logic should be in the use case.

Reply:
Moved the app logic to usecase